cursos/helm/02_helm_charts/02_practica_guiada_meiga.md

```diff
@@ -92,10 +131,86 @@ Y cambiamos la línea correnpondiente de `frontend-service.yaml`
  - protocol: TCP
     port: {{ .Values.puertos.servicio.frontend }}
     targetPort: 80
+...
+```
+
+### b) Imágenes como parámetro
+Vamos a mover a **values** las imágenes que utilizamos tanto en *bbdd* como en *frontend* para así poder cambiar la imagen o su versión sin tener que tocar ningún artefacto.
+
+`values.yaml`
+```yaml
+...
+imagenes: 
+  frontend: elberto/meiga-php-lite
+  bbdd: mysql:5.7
+```
+
+`bbdd-deploy` (lo mismo para `frontend-deploy`)
+```yaml
+...
+containers:
+      - name: meiga-mysql
+        image: {{ .Values.imagenes.bbdd }}
+...
+```
+
+### c) Migrar los valores de nuestro configmap a Values
+En el `archivo configmap.yaml` tenemos la siguiente información:
+```yaml
+...
+  CURSO: "nomeCurso"
+  DOCENTE: "nomeAlumno"
+  MYSQL_HOST: servizo-{{ .Release.Name }}-bbdd
+  MYSQL_USER: "root"
+  MYSQL_DATABASE: "meiga"
+```
+
+Dejaremos `MYSQL_HOST` pues no tiene mucho sentido exponerlo para su configuración. Vamos a declarar estas variables en `values.yaml`.
 ```
+#values.yaml
+...
+env:
+  CURSO: "Aprende Helm"
+  DOCENTE: "Estudiante1"
 
-Vamos a mover a values las imágenes que utilizamos tanto en *bbdd* como en *frontend*, así como todos los valores del *configmap* y *secret*.
+mysql:
+  user: "root"
+  database: "meiga"
+```
 
+Modificaremos el `configmap.yaml` de la siguiente manera:
+```yaml
+data:
+  MYSQL_HOST: servizo-{{ .Release.Name }}-bbdd
+  MYSQL_USER: {{ .Values.mysql.user }}
+  MYSQL_DATABASE: {{ .Values.mysql.database }}
+  # En vez de coger una a una todas las variables de entorno que modifican 
+  # la aplicacion, las definimos en Values.env. Si hace falta añadir alguna
+  # no tenemos que tocar el artefacto, solo añadir una propiedad en Values.env
+  {{ .Values.env | toYaml | nindent 2}}
+```
+
+
+
+### d) Contenido del secrets
+Vamos a exponer la contraseña de nuestra base de datos para que se pueda modificar desde `values.yaml`:
+```yaml
+#values.yaml
+...
+secretos:
+  rootpass: contrasinal
+```
+
+Vamos a utilizar una de las funciones que nos proporciona [Sprig](http://masterminds.github.io/sprig/) para introducir como texto nuestra contraseña en `values.yaml` y que Helm la inyecte dentro de `secret.yaml` en formato base64:
```

Comment:
@AlbertoFemenias We should create a section for introduce the pipes and the function system of Helm's sprig. Should it be a new issue/task?

Reply:
I added yesterday issue #44 with some concepts I thought should be included, I will add "pipes and functions in Helm"